### PR TITLE
Let (almost) all flutter create CocoaPods references be dynamically generated later

### PR DESCRIPTION
**Pull request by xster**
Yolo #9763

Remove everything from the flutter create template (except Podfile, next task in #9763).
Refer only to generated CocoaPods files from our own generated xcconfig files :)
pubspec is the single source of truth and xcodebuild errors should always accompany our own warning messages

Should achieve:
- Fresh project works on machine without CocoaPods.
- Building project doesn't dump any transient generated files in source controlled files.
- Source controlling a fresh project and cloning onto another machine without CocoaPods works.
- Installing CocoaPods and adding a plugin works and building works (and more or less reverts this, but on the user's local machine).
- Checking in the project after adding a plugin forces other developers on the same team to install CocoaPods with a message.
- Project diff after adding a plugin is reasonable:

```
diff --git a/ios/Runner.xcodeproj/project.pbxproj b/ios/Runner.xcodeproj/project.pbxproj
index 8bf0b92..df3d27c 100644
--- a/ios/Runner.xcodeproj/project.pbxproj
+++ b/ios/Runner.xcodeproj/project.pbxproj
@@ -7,6 +7,7 @@
        objects = {

 /* Begin PBXBuildFile section */
+               0BC75B219FF778F6C3028D0F /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = E81B7E679E02EF886E073688 /* Pods_Runner.framework */; };
                1498D2341E8E89220040F4C2 /* PluginRegistry.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* PluginRegistry.m */; };
                3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
                3B80C3941E831B6300D905FE /* App.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 3B80C3931E831B6300D905FE /* App.framework */; };
@@ -56,6 +57,7 @@
                97C146FD1CF9000F007C117D /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
                97C147001CF9000F007C117D /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.storyboard; name = Base; path = Base.lproj/LaunchScreen.storyboard; sourceTree = "<group>"; };
                97C147021CF9000F007C117D /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
+               E81B7E679E02EF886E073688 /* Pods_Runner.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_Runner.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 /* End PBXFileReference section */

 /* Begin PBXFrameworksBuildPhase section */
@@ -65,6 +67,7 @@
                        files = (
                                9705A1C61CF904A100538489 /* Flutter.framework in Frameworks */,
                                3B80C3941E831B6300D905FE /* App.framework in Frameworks */,
+                               0BC75B219FF778F6C3028D0F /* Pods_Runner.framework in Frameworks */,
                        );
                        runOnlyForDeploymentPostprocessing = 0;
                };
@@ -91,6 +94,8 @@
                                9740EEB11CF90186004384FC /* Flutter */,
                                97C146F01CF9000F007C117D /* Runner */,
                                97C146EF1CF9000F007C117D /* Products */,
+                               DD82CC3E7AAF1538644011F8 /* Pods */,
+                               B375B80D59CCF6A9DFA06DC6 /* Frameworks */,
                        );
                        sourceTree = "<group>";
                };
@@ -126,6 +131,21 @@
                        name = "Supporting Files";
                        sourceTree = "<group>";
                };
+               B375B80D59CCF6A9DFA06DC6 /* Frameworks */ = {
+                       isa = PBXGroup;
+                       children = (
+                               E81B7E679E02EF886E073688 /* Pods_Runner.framework */,
+                       );
+                       name = Frameworks;
+                       sourceTree = "<group>";
+               };
+               DD82CC3E7AAF1538644011F8 /* Pods */ = {
+                       isa = PBXGroup;
+                       children = (
+                       );
+                       name = Pods;
+                       sourceTree = "<group>";
+               };
 /* End PBXGroup section */

 /* Begin PBXNativeTarget section */
@@ -133,12 +153,15 @@
                        isa = PBXNativeTarget;
                        buildConfigurationList = 97C147051CF9000F007C117D /* Build configuration list for PBXNativeTarget "Runner" */;
                        buildPhases = (
+                               C4C659CECFF01A9A9C084B1A /* [CP] Check Pods Manifest.lock */,
                                9740EEB61CF901F6004384FC /* Run Script */,
                                97C146EA1CF9000F007C117D /* Sources */,
                                97C146EB1CF9000F007C117D /* Frameworks */,
                                97C146EC1CF9000F007C117D /* Resources */,
                                9705A1C41CF9048500538489 /* Embed Frameworks */,
                                3B06AD1E1E4923F5004D2608 /* Thin Binary */,
+                               279674D47E765FE4C991DFFA /* [CP] Embed Pods Frameworks */,
+                               B5485F1AF24C3FECA90741DD /* [CP] Copy Pods Resources */,
                        );
                        buildRules = (
                        );
@@ -200,6 +223,21 @@
 /* End PBXResourcesBuildPhase section */

 /* Begin PBXShellScriptBuildPhase section */
+               279674D47E765FE4C991DFFA /* [CP] Embed Pods Frameworks */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Embed Pods Frameworks";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
+                       shellPath = /bin/sh;
+                       shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh\"\n";
+                       showEnvVarsInLog = 0;
+               };
                3B06AD1E1E4923F5004D2608 /* Thin Binary */ = {
                        isa = PBXShellScriptBuildPhase;
                        buildActionMask = 2147483647;
@@ -228,6 +266,36 @@
                        shellPath = /bin/sh;
                        shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" build";
                };
+               B5485F1AF24C3FECA90741DD /* [CP] Copy Pods Resources */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Copy Pods Resources";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
                                97C146EA1CF9000F007C117D /* Sources */,
                                97C146EB1CF9000F007C117D /* Frameworks */,
                                97C146EC1CF9000F007C117D /* Resources */,
                                9705A1C41CF9048500538489 /* Embed Frameworks */,
                                3B06AD1E1E4923F5004D2608 /* Thin Binary */,
+                               279674D47E765FE4C991DFFA /* [CP] Embed Pods Frameworks */,
+                               B5485F1AF24C3FECA90741DD /* [CP] Copy Pods Resources */,
                        );
                        buildRules = (
                        );
@@ -200,6 +223,21 @@
 /* End PBXResourcesBuildPhase section */

 /* Begin PBXShellScriptBuildPhase section */
+               279674D47E765FE4C991DFFA /* [CP] Embed Pods Frameworks */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Embed Pods Frameworks";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
+                       shellPath = /bin/sh;
+                       shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh\"\n";
+                       showEnvVarsInLog = 0;
+               };
                3B06AD1E1E4923F5004D2608 /* Thin Binary */ = {
                        isa = PBXShellScriptBuildPhase;
                        buildActionMask = 2147483647;
@@ -228,6 +266,36 @@
                        shellPath = /bin/sh;
                        shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" build";
                };
+               B5485F1AF24C3FECA90741DD /* [CP] Copy Pods Resources */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Copy Pods Resources";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
diff --git a/ios/Runner.xcodeproj/project.pbxproj b/ios/Runner.xcodeproj/project.pbxproj
index 8bf0b92..df3d27c 100644
--- a/ios/Runner.xcodeproj/project.pbxproj
+++ b/ios/Runner.xcodeproj/project.pbxproj
@@ -7,6 +7,7 @@
        objects = {

 /* Begin PBXBuildFile section */
+               0BC75B219FF778F6C3028D0F /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = E81B7E679E02EF886E073688 /* Pods_Runner.framework */; };
                1498D2341E8E89220040F4C2 /* PluginRegistry.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* PluginRegistry.m */; };
                3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
                3B80C3941E831B6300D905FE /* App.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 3B80C3931E831B6300D905FE /* App.framework */; };
@@ -56,6 +57,7 @@
                97C146FD1CF9000F007C117D /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
                97C147001CF9000F007C117D /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.storyboard; name = Base; path = Base.lproj/LaunchScreen.storyboard; sourceTree = "<group>"; };
                97C147021CF9000F007C117D /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
+               E81B7E679E02EF886E073688 /* Pods_Runner.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_Runner.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 /* End PBXFileReference section */

 /* Begin PBXFrameworksBuildPhase section */
@@ -65,6 +67,7 @@
                        files = (
                                9705A1C61CF904A100538489 /* Flutter.framework in Frameworks */,
                                3B80C3941E831B6300D905FE /* App.framework in Frameworks */,
+                               0BC75B219FF778F6C3028D0F /* Pods_Runner.framework in Frameworks */,
                        );
                        runOnlyForDeploymentPostprocessing = 0;
                };
@@ -91,6 +94,8 @@
                                9740EEB11CF90186004384FC /* Flutter */,
                                97C146F01CF9000F007C117D /* Runner */,
                                97C146EF1CF9000F007C117D /* Products */,
+                               DD82CC3E7AAF1538644011F8 /* Pods */,
+                               B375B80D59CCF6A9DFA06DC6 /* Frameworks */,
                        );
                        sourceTree = "<group>";
                };
@@ -126,6 +131,21 @@
                        name = "Supporting Files";
                        sourceTree = "<group>";
                };
+               B375B80D59CCF6A9DFA06DC6 /* Frameworks */ = {
+                       isa = PBXGroup;
+                       children = (
+                               E81B7E679E02EF886E073688 /* Pods_Runner.framework */,
+                       );
+                       name = Frameworks;
+                       sourceTree = "<group>";
+               };
+               DD82CC3E7AAF1538644011F8 /* Pods */ = {
+                       isa = PBXGroup;
+                       children = (
+                       );
+                       name = Pods;
+                       sourceTree = "<group>";
+               };
 /* End PBXGroup section */

 /* Begin PBXNativeTarget section */
@@ -133,12 +153,15 @@
                        isa = PBXNativeTarget;
                        buildConfigurationList = 97C147051CF9000F007C117D /* Build configuration list for PBXNativeTarget "Runner" */;
                        buildPhases = (
+                               C4C659CECFF01A9A9C084B1A /* [CP] Check Pods Manifest.lock */,
                                9740EEB61CF901F6004384FC /* Run Script */,
                                97C146EA1CF9000F007C117D /* Sources */,
                                97C146EB1CF9000F007C117D /* Frameworks */,
                                97C146EC1CF9000F007C117D /* Resources */,
                                9705A1C41CF9048500538489 /* Embed Frameworks */,
                                3B06AD1E1E4923F5004D2608 /* Thin Binary */,
+                               279674D47E765FE4C991DFFA /* [CP] Embed Pods Frameworks */,
+                               B5485F1AF24C3FECA90741DD /* [CP] Copy Pods Resources */,
                        );
                        buildRules = (
                        );
@@ -200,6 +223,21 @@
 /* End PBXResourcesBuildPhase section */

 /* Begin PBXShellScriptBuildPhase section */
+               279674D47E765FE4C991DFFA /* [CP] Embed Pods Frameworks */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Embed Pods Frameworks";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
+                       shellPath = /bin/sh;
+                       shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh\"\n";
+                       showEnvVarsInLog = 0;
+               };
                3B06AD1E1E4923F5004D2608 /* Thin Binary */ = {
                        isa = PBXShellScriptBuildPhase;
                        buildActionMask = 2147483647;
@@ -228,6 +266,36 @@
                        shellPath = /bin/sh;
                        shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" build";
                };
+               B5485F1AF24C3FECA90741DD /* [CP] Copy Pods Resources */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Copy Pods Resources";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
+                       shellPath = /bin/sh;
+                       shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-Runner/Pods-Runner-resources.sh\"\n";
+                       showEnvVarsInLog = 0;
+               };
+               C4C659CECFF01A9A9C084B1A /* [CP] Check Pods Manifest.lock */ = {
+                       isa = PBXShellScriptBuildPhase;
+                       buildActionMask = 2147483647;
+                       files = (
+                       );
+                       inputPaths = (
+                       );
+                       name = "[CP] Check Pods Manifest.lock";
+                       outputPaths = (
+                       );
+                       runOnlyForDeploymentPostprocessing = 0;
+                       shellPath = /bin/sh;
+                       shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n";
+                       showEnvVarsInLog = 0;
+               };
 /* End PBXShellScriptBuildPhase section */

 /* Begin PBXSourcesBuildPhase section */
diff --git a/ios/Runner.xcworkspace/contents.xcworkspacedata b/ios/Runner.xcworkspace/contents.xcworkspacedata
index 1d526a1..21a3cc1 100644
--- a/ios/Runner.xcworkspace/contents.xcworkspacedata
+++ b/ios/Runner.xcworkspace/contents.xcworkspacedata
@@ -4,4 +4,7 @@
    <FileRef
       location = "group:Runner.xcodeproj">
    </FileRef>
+   <FileRef
+      location = "group:Pods/Pods.xcodeproj">
+   </FileRef>
 </Workspace>
diff --git a/pubspec.yaml b/pubspec.yaml
index bff1dcd..51b3927 100644
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -4,6 +4,7 @@ description: A new flutter project.
 dependencies:
   flutter:
     sdk: flutter
+  url_launcher: ^0.3.0
```

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -141,7 +141,12 @@ class CreateCommand extends FlutterCommand {
     printStatus('Wrote $generatedCount files.');
     printStatus('');
 
-    updateXcodeGeneratedProperties(appPath, BuildMode.debug, flx.defaultMainPath);
+    updateXcodeGeneratedProperties(
+      appPath,
+      BuildMode.debug,
+      flx.defaultMainPath,
+      false, // No plugins or CocoaPods by default.
```
Comment (collinjackson) [inline packages/flutter_tools/lib/src/commands/create.dart]: perhaps the arguments should be named so this boolean doesn't require a comment

**Review by collinjackson — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (xster) [inline packages/flutter_tools/lib/src/commands/create.dart]: sure

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 2]
```diff
@@ -141,7 +141,12 @@ class CreateCommand extends FlutterCommand {
     printStatus('Wrote $generatedCount files.');
     printStatus('');
 
-    updateXcodeGeneratedProperties(appPath, BuildMode.debug, flx.defaultMainPath);
+    updateXcodeGeneratedProperties(
+      projectPath: appPath,
+      mode: BuildMode.debug,
+      target: flx.defaultMainPath,
+      hasPlugins: false,
```
Comment (jakobr-google) [inline packages/flutter_tools/lib/src/commands/create.dart]: Should be `hasPlugins: generatePlugin,`. The example app for a plugin project does have plugins. :)

File: packages/flutter_tools/lib/src/ios/xcodeproj.dart [hunk 3]
```diff
@@ -45,6 +52,11 @@ void updateXcodeGeneratedProperties(String projectPath, BuildMode mode, String t
     localsBuffer.writeln('LOCAL_ENGINE=${localEngineArtifacts.engineOutPath}');
   }
 
+  // If the project isn't using platform plugins, don't assume CocoaPods will be
```
Comment (jakobr-google) [inline packages/flutter_tools/lib/src/ios/xcodeproj.dart]: It's correct, but a double-negative is hard to parse. Maybe "Only include CocoaPods dependency if the project is using plugins."?

File: packages/flutter_tools/lib/src/ios/xcodeproj.dart [hunk 4]
```diff
@@ -45,6 +52,11 @@ void updateXcodeGeneratedProperties(String projectPath, BuildMode mode, String t
...
+  // needed and depended on.
+  if (hasPlugins)
+    localsBuffer.writeln('#include "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"');
```
Comment (jakobr-google) [inline packages/flutter_tools/lib/src/ios/xcodeproj.dart]: It'll probably work, but before this we included the debug xcconfig in a debug build.

**Review by jakobr-google — COMMENTED** [review]
LGTM, but please update the create code and check that a newly created plugin project still works.

File: packages/flutter_tools/lib/src/ios/xcodeproj.dart (hunk 3 above)
Comment (xster) [inline packages/flutter_tools/lib/src/ios/xcodeproj.dart]: Done

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 2 above)
Comment (xster) [inline packages/flutter_tools/lib/src/commands/create.dart]: Ah, totally didn't know. Thanks for the catch

File: packages/flutter_tools/lib/src/ios/xcodeproj.dart (hunk 4 above)
Comment (xster) [inline packages/flutter_tools/lib/src/ios/xcodeproj.dart]: ya, I noticed that too. Though it seems like we hardcode the Release configuration in the xcodebuild command anyway
https://github.com/flutter/flutter/blob/b8b6a14d8b1629b24278a01e3dd5ffc06f0756f4/packages/flutter_tools/lib/src/ios/mac.dart#L166

**Comment by xster** [comment]
Thanks for the review. 

Confirmed that with flutter create --plugin
- Without cocoapods installed, trying to run the example gives a warning and fails
- With cocoapods installed, everything hooks up and runs correctly
